lib/features/mailbox_dashboard/presentation/controller/advanced_filter_controller.dart

```diff
@@ -440,38 +492,41 @@ class AdvancedFilterController extends BaseController {
   void _handleClearAllFieldOfAdvancedSearch() {
     _resetAllToOriginalValue();
     _clearAllTextFieldInput();
+    _memorySearchFilter = SearchEmailFilter.initial();
   }
 
-  void _updateFromField() {
-    if (searchEmailFilter.from.isNotEmpty) {
-      listFromEmailAddress = List.from(
-        searchEmailFilter.from.map((address) => EmailAddress(null, address)));
-    } else {
-      listFromEmailAddress.clear();
-    }
+  void onSearchAction() {
+    FocusManager.instance.primaryFocus?.unfocus();
+    applyAdvancedSearchFilter();
   }
 
-  void _updateToField() {
-    if (searchEmailFilter.to.isNotEmpty) {
-      listToEmailAddress = List.from(
-        searchEmailFilter.to.map((address) => EmailAddress(null, address)));
-    } else {
-      listToEmailAddress.clear();
-    }
+  void _handleStartSearchEmailAction() {
+    _memorySearchFilter = searchController.searchEmailFilter.value;
+    initSearchFilterField(currentContext);
   }
 
-  void onSearchAction() {
-    FocusManager.instance.primaryFocus?.unfocus();
-    applyAdvancedSearchFilter();
+  void onHasAttachmentCheckboxChanged(bool? isChecked) {
+    hasAttachment.value = isChecked ?? false;
+    _updateMemorySearchFilter(hasAttachmentOption: Some(hasAttachment.value));
   }
 
-  void _handleSearchEmailBySearchFilterAction(QuickSearchFilter searchFilter) {
-    switch(searchFilter) {
-      case QuickSearchFilter.from:
-        _updateFromField();
+  void onTextChanged(AdvancedSearchFilterField filterField, String value) {
```

Comment:
Please write unit test for this method

Reply:
What is the purpose?

Reply:
idem

Reply:
- Confirming _updateMemorySearchFilter is called with expected values when
  - [ ] `filterField` is `AdvancedSearchFilterField.subject`
  - [ ] `filterField` is `AdvancedSearchFilterField.hasKeyword`
  - [ ] `filterField` is `AdvancedSearchFilterField.notKeyword`
- [ ] Confirming nothing change when `filterField` is none of the above

Reply:
done